server/tserver/src/main/java/org/apache/accumulo/tserver/session/SessionManager.java

```diff
@@ -458,28 +458,36 @@ public List<ActiveScan> getActiveScans() {
         addActiveScan(activeScans, scanSession,
             isSingle ? ((SingleScanSession) scanSession).extent
                 : ((MultiScanSession) scanSession).threadPoolExtent,
-            ct, isSingle ? ScanType.SINGLE : ScanType.BATCH,
-            computeScanState(scanSession.getScanTask()), scanSession.scanParams, entry.getKey());
+            ct, isSingle ? ScanType.SINGLE : ScanType.BATCH, computeScanState(scanSession),
+            scanSession.scanParams, entry.getKey());
       }
     }));
 
     return activeScans;
   }
 
-  private ScanState computeScanState(ScanTask<?> scanTask) {
+  private ScanState computeScanState(ScanSession<?> session) {
     ScanState state = ScanState.RUNNING;
 
-    if (scanTask == null) {
+    if (session.getScanTask() == null) {
       state = ScanState.IDLE;
     } else {
-      switch (scanTask.getScanRunState()) {
+      switch (session.getScanTask().getScanRunState()) {
         case QUEUED:
           state = ScanState.QUEUED;
           break;
         case FINISHED:
           state = ScanState.IDLE;
           break;
         case RUNNING:
+          if (zombieCountConsumer != null) {
+            state = ScanState.ZOMBIE;
+            handleZombieScan(session);
+          } else if (session.getScanTask() == null) {
```

Comment:
```suggestion
          } else if (deferredCleanupQueue.contains(session)) {
```
I believe the CLEANING state is meant for sessions that haven't been cleaned up yet (aka they have been deferred for cleanup)